src/Core/Extensible.php

```diff
@@ -553,8 +557,21 @@ public function getExtensionInstances()
 
             if ($extensions) {
                 foreach ($extensions as $extension) {
-                    $instance = Injector::inst()->get($extension);
-                    $this->extension_instances[get_class($instance)] = $instance;
+                    $name = $extension;
+                    // Allow service names of the form "%$ServiceName"
+                    if (substr($name, 0, 2) == '%$') {
+                        $name = substr($name, 2);
+                    }
+
+                    if (strstr($name, '(')) {
+                        list($name) = ClassInfo::parse_class_spec($name);
```

Comment:
```php
$name = strtok($name, '(');
```

would be a lot faster; You can even skip the `strstr` as it's safe.

Reply:
done